### ISSUE TITLE
Custom types in queries

### ISSUE THREAD

**Issue by cherrydev** [greenrobot/greenDAO#610]
I was surprised there wasn't an issue open (and possibly closed) about this. So, as [noted in the documentation](http://greenrobot.org/greendao/documentation/custom-types/), custom types cannot be used as-is with the querybuilder and must be converted to a primitive type.

I don't understand the reasoning for this as it seems, at least on the surface, to be straightforward to implement. I might go about it by adding an optional converter instance to the Property class, which would give the QueryBuilder a logical place to fetch the converter.

Assuming there is a really good reason for not implementing this, though, I would suggest that the QueryBuilder at LEAST do a basic check to make sure that the value being presented to a WHERE or ORDER clause is actually a supported primitive type. Not doing this has lead to me (and probably many others) having mysteriously failing queries that take a `QueryBuilder.SQL_VALUE = true` to debug.

**Comment by greenrobot-team** [comment]
I suppose this was left out for simplicity, but only @greenrobot can talk to details about this.

Anyhow, you can simply use the converter yourself when building a query: 
```java
RoleConverter converter = new RoleConverter();
List<User> authors = userDao.queryBuilder()
    .where(Properties.Role.eq(converter.convertToDatabaseValue(Role.AUTHOR)))
    .list();
```
-ut

**Comment by cherrydev** [comment]
Yep, I'm totally aware of ways around it and I know that simplicity is important in the greendao project. But I'm wondering if perhaps this omission is a bit TOO simple for the potential effort it would take. If I got the go-ahead I'd be happy to give the implementation a shot.
